File: src/main/java/com/github/qky666/selenidepom/java/Widget.java
package com.github.qky666.selenidepom.java;

import com.codeborne.selenide.SelenideElement;

import javax.annotation.CheckReturnValue;

public abstract class Widget implements RequiredContainer {
    public final SelenideElement self;

    @CheckReturnValue
    public Widget(SelenideElement self) {
        this.self = self;
    }
}

File: src/main/kotlin/com/github/qky666/selenidepom/kotlin/Widget.kt
package com.github.qky666.selenidepom.kotlin

import com.codeborne.selenide.SelenideElement

/**
 * Abstract class that implements [RequiredContainer] interface that has a [SelenideElement] 'self' field.
 */
abstract class Widget (open val self: SelenideElement) : RequiredContainer
